Support getting and setting multiple outputs at once

sirhcel opened the pull request:
The JSON request may affect all outputs and the JSON response always contains information for all outputs. So it would be nice to have the option to set and get for multiple outputs at once.

I made this additions while porting an older command line interface onto PyNetio. But they seem generally useful to me.

In addition, b2dd071 fixes an off-by-one issue betwen `set_output` and `get_output`. The former takes the ID as the ID in the JSON data (starting at one). The latter as index to the response list (starting at zero). The former behaviour seems more intuitive to me, so I went for this one. I hope your are O.K. with this drive-by fix without opening an issue.

AdamVerner commented:
Thanks for the request.

You're right, i was trying to index the ID from zero (Netio outputs are indexed from 1) but that got back at me. 

The reason why `Device` and `JsonDevice` are separated is that `Device` should provide access to outputs and all the interface needed, while `JsonDevice` should care only about data access.

I've merged this request into branch [devel](https://github.com/netioproducts/PyNetio/tree/devel) check that out.

Also i removed the get_outputs filtering as i don't see any advantage there. Do you have any use-case for that? I'd normally use the api somehow like this:
```
r = n.get_outputs()
print(f'output 1 {n[0].State}')
print(f'output 2 {n[1].State}')
```
As the response is always sorted. 
Maybe we could add abstraction to the returned value, where we would have getter by ID (maybe override the `operator[]`), but i feel like that's over-engineering

sirhcel commented:
> I've merged this request into branch [devel](https://github.com/netioproducts/PyNetio/tree/devel) check that out.

Thank you very much. I edited and rebased https://github.com/netioproducts/PyNetio/pull/3 and it works like a treat.

> Also i removed the get_outputs filtering as i don't see any advantage there. Do you have any use-case for that?

I added the parametrized `get_outputs(ids)` with the CLI tool im mind. `get_outputs()` is absolutely fine. You have to get hold on/select the individual entries for further processing anyway and at this point in time filtering by `ID` comes into the play again. Now the CLI tool performs the filtering.

> Maybe we could add abstraction to the returned value, where we would have getter by ID (maybe override the `operator[]`), but i feel like that's over-engineering

I agree with you. This feels to me "too much" for now. `get_outputs()` is straightforward so we could leave this for version 2.0. ;-)

If you don't have any concerns - let's roll.

AdamVerner commented:
I'll merge this into master through devel, once we're done with #3 

This version is incompatible with previous(due to b2dd071). 
We'll have to bump up the MAJOR revsion
